### PR TITLE
#CP-40 chore: set up dotenv

### PR DESCRIPTION
**Pull request by Yvad60**
#### What the PR does
set up dotenv  for better management of environment variables

#### Description of Tasks to be completed

- [x] Moving to production switches to the correct environment
- [x] Shared variables are managed using dotenv
- [x] Sensitive info like secret keys are hidden from the public

#### How this should be manually tested
Clone the project repository linked [here](https://github.com/atlp-rwanda/phantom-be-codebandits)

```
cd phantom-be-codebandits/
git ch-setup-dotenv-cp-40
git pull origin ch-setup-dotenv-cp-40
npm install
create a .env file and add a PORT variable 
npm run start
run http://localhost:yourCreatedPort  in postman or in  your browser
test different routes 
- http://localhost:5000
- http://localhost:5000/api/v1
- http://localhost:5000/docs
- http://localhost:5000/ + any other endpoint
test logging with winston
```

#### Relevant Trello Card
[#CP-40](https://trello.com/c/dLVpMTSl/40-setup-dotenv)

### CODE REVIEW

**Review by feyton — CHANGES_REQUESTED** [review]
@Yvad60 This is great work.
The configurations are working on my side. 

Add the requirement in the `README` file in the installation part.

Add a new entry about creating the `.env` file and we will keep it updated with the requirements as we grow the APP

File: package.json [hunk 1]
```diff
@@ -23,13 +23,14 @@
 	},
 	"homepage": "https://github.com/atlp-rwanda/phantom-be-codebandits#readme",
 	"devDependencies": {
+		"dotenv": "^16.0.0",
```
Comment (Jkadhuwa) [inline package.json]: Make it a prod dependency

**Review by Jkadhuwa — CHANGES_REQUESTED** [review]
Nice work @Yvad60, left some feedback. 

- create a .env.sample  file(do not add it to .gitignore)  and put the needed variables example  e.g  HEROKU_API_KEY=

**Comment by feyton** [comment]
> Nice work @Yvad60, left some feedback.
> 
>     * create a .env.sample  file(do not add it to .gitignore)  and put the needed variables example  e.g  HEROKU_API_KEY=

**Review by feyton — APPROVED** [review]

**Review by Jkadhuwa — APPROVED** [review]

**Pull request merged**